sb3_contrib/common/maskable/algorithms/on_policy.py

```diff
@@ -0,0 +1,132 @@
+            self.device,
+            gamma=self.gamma,
+            gae_lambda=self.gae_lambda,
+            n_envs=self.n_envs,
+        )
+
+        if not isinstance(self.policy, MaskablePolicy):
+            logger.warn("Algorithm's policy does not support invalid action masking")
+
+    def collect_rollouts(
+        self,
+        env: VecEnv,
+        callback: BaseCallback,
+        rollout_buffer: RolloutBuffer,
+        n_rollout_steps: int,
+    ) -> bool:
+        """
+        Collect experiences using the current policy and fill a ``RolloutBuffer``.
+        The term rollout here refers to the model-free notion and should not
+        be used with the concept of rollout used in model-based RL or planning.
+
+        This method is largely identical to the implementation found in the parent class.
+
+        :param env: The training environment
+        :param callback: Callback that will be called at each step
+            (and at the beginning and end of the rollout)
+        :param rollout_buffer: Buffer to fill with rollouts
+        :param n_steps: Number of experiences to collect per environment
+        :return: True if function returned with at least `n_rollout_steps`
+            collected, False if callback terminated rollout prematurely.
+        """
+
+        assert isinstance(rollout_buffer, MaskableRolloutBuffer), "RolloutBuffer doesn't support action masking"
+        assert self._last_obs is not None, "No previous observation was provided"
+        n_steps = 0
+        action_masks = None
+        rollout_buffer.reset()
+        # Sample new weights for the state dependent exploration
+        if self.use_sde:
+            self.policy.reset_noise(env.num_envs)
+
+        callback.on_rollout_start()
+
+        while n_steps < n_rollout_steps:
+            if self.use_sde and self.sde_sample_freq > 0 and n_steps % self.sde_sample_freq == 0:
+                # Sample a new noise matrix
+                self.policy.reset_noise(env.num_envs)
+
+            with th.no_grad():
+                # Convert to pytorch tensor
+                obs_tensor = th.as_tensor(self._last_obs).to(self.device)
+
+                # This is the only change related to invalid action masking
+                if is_vecenv_wrapped(env, VecActionMasker):
+                    action_masks = env.valid_actions()
+
+                if isinstance(self.policy, MaskablePolicy):
+                    actions, values, log_probs = self.policy.forward(obs_tensor, action_masks=action_masks)
+                else:
+                    actions, values, log_probs = self.policy.forward(obs_tensor)
+
+            actions = actions.cpu().numpy()
+
+            # Rescale and perform action
+            clipped_actions = actions
+            # Clip the actions to avoid out of bound error
+            if isinstance(self.action_space, gym.spaces.Box):
+                clipped_actions = np.clip(actions, self.action_space.low, self.action_space.high)
```

Comment:
Saw some code for the continuous action space as well. However, it seems the current implementation only handles the discrete action space. Also I have no idea how to implement invalid action masking with continuous action space haha.

Maybe consider throwing an error if the action space is continuous?